sktime/param_est/base.py

```diff
@@ -396,8 +396,8 @@ def _get_fitted_params(self):
         """
         # default retrieves all self attributes ending in "_"
         # and returns them with keys that have the "_" removed
-        fitted_params = [attr for attr in dir(self) if attr.endswith("_")]
-        fitted_params = [x for x in fitted_params if not x.startswith("_")]
-        fitted_param_dict = {p[:-1]: getattr(self, p) for p in fitted_params}
-
-        return fitted_param_dict
+        return {
+            attr[:-1]: getattr(self, attr)
+            for attr in dir(self)
+            if attr.endswith("_") and not attr.startswith("_")
+        }
```

Comment:
Note to reviewer: This is just a very small refactor to make this piece of code easier to follow and a bit more efficient

Reply:
hm, I forgot this was there. Good spot.

Because there is some separate discussion to have here - if this does not change logic, would you be so kind and move this refactor to a separate PR?

To give some explanation:

* in my internal architectural model, this actually lives in `BaseEstimator`. Quick inspection indeed indicates that the parent base class `BaseEstimator` seems to have an alternative, perhaps better version of the code.
* this would suggest instead the action to delete this code, and potentially refactor the `BaseEstimator` code instead
* the explanation for the duplication might be, we first introduced `get_fitted_params` for parameter fitters, but then realized it makes sense in all estimators.
* mid-term, the plan was to use `scikit-base` `BaseEstimator` to get `get_fitted_params  functionality, where the code *also* should live, in the `BaseEstimator` there. So, a mid-term action would be to remove duplicated code in `sktime` `BaseEstimator` and check `scikit-base` whether it should get an update.

So, to summarize, may I suggest to separate this refactor out in a PR, and make a note in an issue to possibly refactor this locus across three locations - `scikit-base` `BaseEstimator`, `sktime` `BaseEstimator`, and `BaseParamFitter`?

Reply:
Reverted these changes and moved them to #5633 (referenced this comment there)